api/src/main/java/com/epam/pipeline/manager/datastorage/providers/azure/AzureStorageHelper.java

```diff
@@ -453,6 +454,28 @@ public BlobContainerClient getBlobContainerClient(final AzureBlobStorage storage
         return getBlobServiceClient(region, credentials).getBlobContainerClient(storage.getPath());
     }
 
+    public String generateSASToken(final AzureBlobStorage dataStorage,
+                                   final List<DataStorageAction> actions,
+                                   final OffsetDateTime expiryTime) {
+        final BlobContainerClient blobContainerClient = getBlobContainerClient(dataStorage);
+        final DataStorageAction dataStorageAction = actions.get(0);
+        Assert.isTrue(actions.size() == 1, "Multiple actions is not supported for AZURE provider");
+
+        final BlobServiceSasSignatureValues values = new BlobServiceSasSignatureValues(expiryTime,
+                buildPermissions(dataStorageAction))
+                .setProtocol(SasProtocol.HTTPS_ONLY)
+                .setContentType("container");
+        addIPRangeToSASValue(values);
+        try {
```

Comment:
instead of try - catch, let's explicitly check the way how we should generate it, if we have manageIdentity configured -   `generateUserDelegationSas`, if storageKey - `generateSas` else throw that we misconfigured